board/stm32h7/board.h

```diff
@@ -0,0 +1,30 @@
+// ///////////////////////////////////////////////////////////// //
+// Hardware abstraction layer for all different supported boards //
+// ///////////////////////////////////////////////////////////// //
+#include "boards/board_declarations.h"
+
+// ///// Board definition and detection ///// //
+#include "drivers/harness.h"
+#include "drivers/fan.h"
+#include "stm32h7/llfan.h"
+#include "stm32h7/llrtc.h"
+#include "drivers/rtc.h"
+#include "boards/red.h"
+
+void detect_board_type(void) {
+  if(!detect_with_pull(GPIOF, 7, PULL_UP) && !detect_with_pull(GPIOF, 8, PULL_UP) && !detect_with_pull(GPIOF, 9, PULL_UP) && !detect_with_pull(GPIOF, 10, PULL_UP)){
```

Comment:
since all H7 pandas are going to use this GPIOF7-10 for identification, let's build a helper which just reads these out and gives a uint8_t or something which we can compare